frontend/src/Patient/PatientComponents/MedicationUploadPageComponents/ConfirmMedicationPhotoButton.jsx

```diff
@@ -1,19 +1,45 @@
 import { useAuth } from '../../../hooks/AuthProvider';
+import { Alert } from 'react-bootstrap'
 import * as API from '../../../api';
+import { useState } from 'react';
 
 function ConfirmMedicationPhotoButton({ medicationInformation, imgSrc }) {
+    const [isFailureAlertShowing, setIsFailureAlertShowing] = useState(false);
+    const [isSuccessAlertShowing, setIsSuccessAlertShowing] = useState(false);
+    const [errorMsg, setErrorMsg] = useState("");
     const { user } = useAuth();
 
+    function handleShowSuccessAlert() {
+        setIsSuccessAlertShowing(true);
+        setTimeout(() => setIsSuccessAlertShowing(false), 3000);
+    }
+
+    function handleShowFailureAlert() {
+        setIsFailureAlertShowing(true);
+        setTimeout(() => setIsFailureAlertShowing(false), 3000);
+    }
+
     async function uploadMedication() {
         try {
             await API.postMedication(user.id, medicationInformation.name, medicationInformation.strength, imgSrc);
+            handleShowSuccessAlert();
         } catch (e) {
+            setErrorMsg(e.message);
+            handleShowFailureAlert();
             console.error(e);
         }
     }
 
     return (
-        <button className="btn btn-success w-100 p-4 mb-3" onClick={uploadMedication}>Confirm</button>
+        <>
+            <button className="btn btn-success w-100 p-4 mb-3" onClick={uploadMedication}>Confirm</button>
+            {isSuccessAlertShowing && <Alert variant='success' onClose={() => setIsSuccessAlertShowing(false)} dismissible className='position-fixed top-0 translate-middle-x mt-3'>
+                <strong>Medication Uploaded Successfully!</strong>
+            </Alert>}
```

Comment:
React Bootstrap's alerts seem to have a `show` param that you can just pass in, did you consider using that here?
```suggestion
             <Alert show={isSuccessAlertShowing} variant='success' onClose={() => setIsSuccessAlertShowing(false)} dismissible className='position-fixed top-0 translate-middle-x mt-3'>
                <strong>Medication Uploaded Successfully!</strong>
            </Alert>
```

Reply:
This is awesome! I didn't see this when going through the documentation, thank you!